packages/flutter/lib/src/http/response.dart

```diff
@@ -2,23 +2,20 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
 import 'dart:typed_data';
 
 /// An HTTP response where the entire response body is known in advance.
 class Response {
   /// Creates a [Response] object with the given fields.
   ///
   /// If [bodyBytes] is non-null, it is used to populate [body].
-  Response({
-    Uint8List bodyBytes,
-    this.statusCode
-  }) : body = bodyBytes != null ? new String.fromCharCodes(bodyBytes) : null,
-       bodyBytes = bodyBytes;
+  Response.bytes(this.bodyBytes, this.statusCode, { this.headers: const {} });
 
   /// The result of decoding [bodyBytes] using ISO-8859-1.
   ///
   /// If [bodyBytes] is null, this will also be null.
-  final String body;
+  String get body => _encodingForHeaders(headers).decode(bodyBytes);
```

Comment:
I'm just matching the API from `package:http`.  I agree that it's a misleading API.